modules/auxiliary/gather/windows_secrets_dump.rb

```diff
@@ -1015,30 +1005,40 @@ def run
       end
     end
 
-    boot_key = ''
-    begin
-      boot_key = get_boot_key if @winreg
-    rescue RubySMB::Error::RubySMBError => e
-      print_error("Error when getting BootKey: #{e}")
-    end
-    if boot_key.empty?
-      if action.name == 'DOMAIN'
-        print_warning('Unable to get BootKey... continuing since action is DOMAIN')
-      else
-        fail_with(Module::Failure::Unknown,
-                  'Unable to get BootKey. If it is a Domain Controller, you can still '\
-                  'try DOMAIN action since it does not need BootKey')
+    unless action.name == 'DOMAIN'
+      boot_key = ''
+      begin
+        boot_key = get_boot_key if @winreg
+      rescue RubySMB::Error::RubySMBError => e
+        if ['DOMAIN', 'ALL'].include?(action.name)
+          print_warning("Error when getting BootKey... skipping: #{e}")
+        else
+          print_error("Error when getting BootKey: #{e}")
+        end
+      end
+      if boot_key.empty?
+        if action.name == 'ALL'
+          print_warning('Unable to get BootKey... skipping')
+        else
+          fail_with(Module::Failure::Unknown,
```

Comment:
```suggestion
          fail_with(Module::Failure::NotFound,
```

NotFound might be more descriptive for this case.